src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,174 @@
+package duke;
+
+/**
+ * Class to keep track of Tasks.
+ */
+public class Duke {
+
+
+
+    private TaskList taskList;
+    private Storage storage;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructor for a Duke.
+     */
+    public Duke() {
+        this.storage = new Storage();
+        this.taskList = this.storage.load();
+        this.ui = new Ui();
+        this.parser = new Parser();
+    }
+
+    /**
+     * Exits the program.
+     */
+    public void exit() {
+        this.ui.showExitMessage();
+        System.exit(0);
+    }
+
+    /**
+     * Adds a task for Duke to track given a parsed command.
+     *
+     * @param t String array of parsed user input.
+     * @throws DukeException Exception for wrong user inputs.
+     */
+    public String add(String[] t) throws DukeException {
+        Task newTask;
+        switch (t[0]) {
+        case "todo":
+            if (t[1].equals("")) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            } else {
+                newTask = new Todo(t);
+            }
+            break;
+        case "deadline":
+            if (t[1].equals("") || t[2].equals("")) {
+                throw new DukeException("The description or deadline of a deadline cannot be empty.");
+            } else {
+                newTask = new Deadline(t);
+            }
+            break;
+        case "event":
+            if (t[1].equals("") || t[2].equals("")) {
+                throw new DukeException("The description or scheduled time of an event cannot be empty.");
+            } else {
+                newTask = new Event(t);
+            }
+            break;
+        default:
+            throw new DukeException("Invalid task. Please check that your input is correct");
+        }
+        this.taskList.addTask(newTask);
+        this.storage.write(this.taskList);
+        return this.ui.showAddMessage(newTask, this.taskList);
+    }
+
+    /**
+     * Gets the task from duke given an index.
+     *
+     * @param index Index of duke.Task.
+     * @return duke.Task of the given index.
+     */
+    public Task getTaskByIndex(int index) {
+        return this.taskList.getTaskByIndex(index);
+    }
+
+    /**
+     * Marks a task as done given its position in the list.
+     *
+     * @param itemNum Position of duke.Task in the list.
+     */
+    public String markDone(int itemNum) {
+        this.taskList.markDone(itemNum);
+        this.storage.write(this.taskList);
+        return this.ui.showMarkDoneMessage(this.getTaskByIndex(itemNum - 1));
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param items Parsed delete command from user.
+     * @throws DukeException Exception for wrong user inputs.
+     */
+    public String deleteTask(String[] items) throws DukeException {
```

Comment:
Any reason why you use an if-else block instead of just an if-block for the guard clause?